### ISSUE TITLE
Cwazy slow on Microsoft Edge

### ISSUE THREAD

**Issue by trokster** [jo/session25519#1]
Hi Jo,
Just dropping a note to say this lib is really slow on microsoft edge ( about 100fold compared to chrome ) not sure why :)

**Comment by jo** [comment]
Hi @trokster,

please read https://github.com/jo/session25519#performance - maybe adding the mentioned `setImmediate` shim helps?

**Comment by grempe** [comment]
Well, its not supposed to be _fast_.  That's kind of the point.  But you didn't say how slow 'Cwazy slow' is. :-)

Deriving a set of session keys for me on OS X Safari (2014 Core i7 MacBook Pro) takes about four seconds roughly.  If you are seeing issues on Edge that are much slower than that I think your issue would have to do with the scrypt implementation and something that Edge doesn't like about it. This code doesn't do anything but run a single hash (which run at the rate of millions/second) and the scrypt key derivation which is supposed to take on the order of seconds to run.

I think this issue would then belong over in script-async-js for @dchest to take a look if you are seeing more than 5 seconds on a laptop class device with the scrypt params we are using.

https://github.com/dchest/scrypt-async-js

I think the `setImmediate` shim is more likely to help perceived performance when you are running other async tasks alongside this, for example a react UI, that you want to feel interactive and not blocking on scrypt running.  It shouldn't really make scrypt run 'faster' as that would kind of defeat the purpose of a memory/cpu hard key derivation function.

Cheers

**Comment by jo** [comment]
@grempe yes you're totally right, `setImmediate` is used to prevent the browser from looking freezed, not to speed things up.

@trokster is the speed difference between ie and chrome also in 1.1.0 that drastic?
